corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -141,42 +160,91 @@ hqDefine("geospatial/js/geospatial_map", [
-                    clickedMarker.remove();
-                }
-                if (draw.getMode() === 'draw_polygon') {
-                    // It's weird moving the marker around with the ploygon
-                    return;
-                }
-                clickedMarker = new mapboxgl.Marker({color: "FF0000", draggable: true});
-                clickedMarker.setLngLat(coordinates);
-                clickedMarker.addTo(map);
-            }
-
             // Handle click events here
             map.on('click', (event) => {
                 let coordinates = getCoordinates(event);
             });
             return self;
         };
 
+        var exportGeoJson = function(drawInstance) {
+            // Credit to https://gist.github.com/danswick/36796153bd86ce982a59043cbe0ac8f7
+            var exportButton = $("#btnExport");
+            var data = drawInstance.getAll();
+
+            if (data.features.length) {
+                // Stringify the GeoJson
+                var convertedData = 'text/json;charset=utf-8,' + encodeURIComponent(JSON.stringify(data));
+
+                // Create export
+                exportButton.attr('href', 'data:' + convertedData);
+                exportButton.attr('download','data.geojson');
+            }
+        };
+
+        var mapControlsModel = function () {
+            'use strict';
+            var self = {};
+            var mapboxinstance = map.getMapboxInstance();
+            var polygons = [];
+            self.btnExportDisabled = ko.observable(true);
+
+            var mapHasPolygons = function() {
+                return polygons.length;
+            };
+
+            function removeId(id) {
+                var index = polygons.indexOf(id); // Find the index of the specified ID
+                polygons.splice(index, 1); // Remove the ID from the array
+              }
+
+            mapboxinstance.on('draw.delete', function(e) {
+                e.features.forEach((feature) => {
+                    if (feature.geometry.type == "Polygon") {
+                        removeId(feature.id);
+                    }
+                });
+                self.btnExportDisabled(!mapHasPolygons());
+            });
+
+            mapboxinstance.on('draw.create', function(e) {
+                e.features.forEach((feature) => {
+                    if (feature.geometry.type == "Polygon") {
+                        polygons.push(feature.id)
+                    }
+                });
+                self.btnExportDisabled(!mapHasPolygons());
+            });
+
+            return self;
+        }
 
         $(document).ajaxComplete(function () {
             // This fires everytime an ajax request is completed
             var mapDiv = $('#geospatial-map');
             var $data = $(".map-data");
+            var $exportButton = $("#btnExport");
 
             if (mapDiv.length && !map) {
                 map = loadMapBox();
             }
 
+            $exportButton.click(function(e) {
```

Comment:
This can be modelled as [click binding](https://knockoutjs.com/documentation/click-binding.html)

Reply:
I tried with knockout, but for some reason a download wasn't triggerd when clicking on the export button. I confirmed that the attributes are in fact being set, but alas.
![Screenshot from 2023-06-06 15-09-46](https://github.com/dimagi/commcare-hq/assets/64970009/913a1db8-1483-4b6b-af8a-56669fd4dc60)